sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -600,6 +600,7 @@ class SparkSession private(
    * @since 2.0.0
    */
   def sql(sqlText: String): DataFrame = {
+    SparkSession.setActiveSession(this)
```

Comment:
To keep it in scope of thriftserver, could we set it in SparkExecuteStatementOperation.execute, using the sqlContext available there?
I'm not sure what other implications it can have here, it would be safer to set there.

Reply:
> To keep it in scope of thriftserver, could we set it in SparkExecuteStatementOperation.execute, using the sqlContext available there?
> I'm not sure what other implications it can have here, it would be safer to set there.

It's ok  to set in SparkExecuteStatementOperation.execute.
https://github.com/apache/spark/pull/26187#discussion_r336877350
I think it ok to set this before parse sql, It was the right thing to do

Reply:
Ok. Let the reviewers of https://github.com/apache/spark/pull/26187 decide it there. (@cloud-fan @zsxwing )

Reply:
So the issue with async queries was that the parser was using the wrong dialect fallback, because it was picking the wrong SQLConf, because it was not set in the background execution thread?
@wangyum @AngersZhuuuu Does fixing this potentially fix any other of the failures in ThriftServerQueryTestSuite that are currently blacklisted to be ignored?

Reply:
@juliuszsompolski Other failures can't be fixed by #26187:

https://github.com/apache/spark/blob/7f8841b7927808632fbe1a8d4b2fa06fe9c85c4d/sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/ThriftServerQueryTestSuite.scala#L87-L102